specs/governance/gov-token.md

```diff
@@ -0,0 +1,93 @@
+# Governance Token
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+  - [Token Minting](#token-minting)
+  - [Token Burning](#token-burning)
+  - [Voting Power](#voting-power)
+    - [Queries](#queries)
+  - [Delegation](#delegation)
+  - [Checkpoints](#checkpoints)
+  - [Overrides](#overrides)
+  - [Supply Cap](#supply-cap)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+| Constant          | Value                                        |
+|-------------------|----------------------------------------------|
+| Address           | `0x4200000000000000000000000000000000000042` |
+| Token name        | `Optimism`                                   |
+| Token symbol      | `OP`                                         |
+
+`GovernanceToken` is an [ERC20](https://eips.ethereum.org/EIPS/eip-20) token contract that inherits from `ERC20Burnable`,
+`ERC20Votes`, and `Ownable`. It allows token holders to delegate their voting power to other addresses, enabling a representative
+voting system.
+
+The specific implementation details of the voting and delegation logic are inherited from the `ERC20Votes` contract. The
+`GovernanceToken` contract focuses on integrating this functionality with the ERC20 token standard and adding the minting
+capability.
+
+### Token Minting
+
+The contract MUST have a `mint` function with external visibility that allows the contract owner to mint new tokens to an
+arbitrary address. This function MUST only be called by the contract owner, as enforced by the `onlyOwner` modifier inherited
+from the `Ownable` contract. When tokens are minted, the voting power of the recipient address MUST be updated accordingly.
+
+### Token Burning
+
+The contract MUST allow token holders to burn their own tokens using the inherited `burn` function from `ERC20Burnable`.
```

Comment:
Is `burn(uint256)` part of the public ABI?

Reply:
yes, because the contract inherits form `ERC20Burnable`, which is where `burn(uint256) is defined: https://github.com/voteagora/governor/blob/main/src/lib/OptimismToken.sol#L550-L552